Sources/Hummingbird/Application.swift

```diff
@@ -134,7 +134,28 @@ public final class HBApplication: HBExtensible {
 
     // MARK: Methods
 
-    /// Run application
+    /// Start application and wait for it to stop
+    ///
+    /// This function can only be called from a non async context as it stalls
+    /// the current thread waiting for the application to finish
+    @available(*, noasync, message: "Use HBApplication.asyncRun instead.")
+    public func run() throws {
+        try self.start()
+        self.wait()
+    }
+
+    /// Start application and wait for it to stop
+    ///
+    /// Version of `run`` that can be called from asynchronous context
+    @available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+    public func asyncRun() async throws {
+        try await self.onExecutionQueue { app in
+            try app.start()
+            app.wait()
```

Comment:
This should preferably use `asyncWait()` instead, right? Code de-duplication

Reply:
Oh you mean
```swift
try app.start()
await app.asyncWait()
```